gnucash/gnome-utils/gnc-file.c

```diff
@@ -640,6 +643,35 @@ gnc_file_query_save (GtkWindow *parent, gboolean can_cancel)
     return TRUE;
 }
 
+static void fix_budget_sign (QofInstance* data, gpointer user_data)
+{
+    GncBudget* budget = GNC_BUDGET(data);
+    const GncGUID *budget_guid = gnc_budget_get_guid (budget);
+    SCM reverse_proc = scm_c_eval_string ("gnc:reverse-budget-signs");
+    SCM scm_budget_guid = scm_from_utf8_string (guid_to_string(budget_guid));
+    scm_call_1 (reverse_proc, scm_budget_guid);
+}
+
+/* For GnuCash 4.0 onwards - fix budget signs */
+/* guard is set if we have completed reversal, or there are no
+   budgets in book */
+static void consider_fix_budget_signs (QofBook *book)
+{
+    if (!(gnc_features_check_used (book, GNC_FEATURE_BUDGET_UNREVERSED)))
```

Comment:
To avoid nesting loops in C you may consider returning early from checks. So instead you'd write

```
if (gnc_features_check_used (book, GNC_FEATURE_BUDGET_UNREVERSED))
    return;
if (!gnc_budget_get_default (book))
    return;
if (...
```